### ISSUE TITLE
JSoup differs from browsers around commented HTML attributes

### ISSUE THREAD

**Issue by panthony** [jhy/jsoup#1938]
Hi,

I encountered a case where JSoup differs from what browsers (Chrome, Firefox Safari) do.

Using this piece of HTML on [try jsoup](https://try.jsoup.org/):

```html
<html>
<head>
<title>Try jsoup</title>
</head>
<body>
  <h1>before</h1>
  <div <!--="" id="hidden" --="">
      <h1>within</h1>
  </div>
   <h1>after</h1>
</body>
</html>
```

Jsoup will produce:

```html
<html>
 <head>
  <title>Try jsoup</title>
 </head>
 <body>
  <h1>before</h1>
  <div>
   <!--="" id="hidden" --="">
      <h1>within</h1>
  </div>
   <h1>after</h1>
</body>
</html>
-->
  </div>
 </body>
</html>
```

Commenting the rest of the body whereas all major navigators will escape the comment character and shows the 3 titles.

**Comment by panthony** [comment]
Probably a similar issue than https://github.com/jhy/jsoup/issues/1483 except here it comment pretty much all the HTML.

**Comment by jhy** [comment]
Thanks - resolved with #1483. Revalidated in browser and jsoup.
